de-lib/src/test/java/org/iplantc/de/tools/client/presenter/ManageToolsViewPresenterTest.java

```diff
@@ -0,0 +1,366 @@
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Created by sriram on 6/7/17.
+ */
+@RunWith(GwtMockitoTestRunner.class)
+public class ManageToolsViewPresenterTest {
+
+    @Mock
+    ManageToolsView toolsViewMock;
+
+    @Mock
+    ManageToolsView.ManageToolsViewAppearance appearanceMock;
+
+    @Mock
+    ToolServices toolServicesMock;
+
+    @Mock
+    AsyncProviderWrapper<EditToolDialog> editDialogProviderMock;
+
+    @Captor
+    ArgumentCaptor<AsyncCallback<EditToolDialog>> editDialogCaptor;
+
+    @Mock
+    EditToolDialog editToolDialogMock;
+
+    @Mock
+    IplantAnnouncer announcerMock;
+
+    @Mock
+    AsyncProviderWrapper<ToolSharingDialog> shareDialogProviderMock;
+
+    @Captor
+    ArgumentCaptor<AsyncCallback<ToolSharingDialog>> sharingDialogCaptor;
+
+    @Mock
+    ToolSharingDialog toolSharingDialogMock;
+
+    @Mock
+    AsyncProviderWrapper<NewToolRequestDialog> newToolRequestDialogProviderMock;
+
+    @Captor
+    ArgumentCaptor<AsyncCallback<NewToolRequestDialog>> newToolDialogCaptor;
+
+    @Mock
+    NewToolRequestDialog toolRequestMock;
+
+    @Mock
+    AsyncProviderWrapper<ToolInfoDialog> toolInfoDialogProviderMock;
+
+    @Captor
+    ArgumentCaptor<AsyncCallback<ToolInfoDialog>> toolInfoDialogCaptor;
+
+    @Mock
+    ToolInfoDialog toolInfoDialogMock;
+
+    @Mock
+    EventBus eventBusMock;
+
+    @Mock
+    List<Tool> currentSelectionMock;
+
+    @Mock
+    ManageToolsToolbarView manageToolsToolbarViewMock;
+
+    @Captor
+    ArgumentCaptor<AppsCallback<List<Tool>>> toolListCaptor;
+
+    @Captor
+    ArgumentCaptor<AppsCallback<List<App>>> appListCaptor;
+
+
+    @Captor
+    ArgumentCaptor<AppsCallback<Tool>> toolCaptor;
+
+
+    @Captor
+    ArgumentCaptor<AppsCallback<Void>> voidCaptor;
+
+    private ManageToolsViewPresenter uut;
+
+    @Mock
+    Iterator<Tool> iteratorMock;
+
+    @Before
+    public void setUp() {
+        uut = new ManageToolsViewPresenter(appearanceMock);
+        uut.toolsView = toolsViewMock;
+        uut.announcer = announcerMock;
+        uut.appearance = appearanceMock;
+        uut.toolServices = toolServicesMock;
+        uut.editDialogProvider = editDialogProviderMock;
+        uut.shareDialogProvider = shareDialogProviderMock;
+        uut.newToolRequestDialogProvider = newToolRequestDialogProviderMock;
+        uut.toolInfoDialogProvider = toolInfoDialogProviderMock;
+        uut.eventBus = eventBusMock;
+        uut.currentSelection = currentSelectionMock;
+    }
+
+    @Test
+    public void testGo() {
+        HasOneWidget containerMock = mock(HasOneWidget.class);
+        when(toolsViewMock.getToolbar()).thenReturn(manageToolsToolbarViewMock);
+
+        SimpleContainer scMock = mock(SimpleContainer.class);
+        when(toolsViewMock.asWidget()).thenReturn(scMock);
+
+        uut.go(containerMock);
+        verify(containerMock).setWidget(eq(scMock));
+        verify(toolsViewMock, times(10)).getToolbar();
```

Comment:
I personally would test each individual handler that gets added to the toolbar instead of testing the number of times the view returns the toolbar.